### ISSUE TITLE
C++: mismatch end of namespace on uncommented code near another namespace

### ISSUE THREAD

**Issue by daktfi** [Wilfred/difftastic#416]
In header there were two commented out declarations with comments at the end of line.
Upon uncommenting those declarations and removing trailing comments difftastic mismatched end of namespace between files with the subsequent namespace. Removing second namespace removes error. Removing trailing comments in "before" does NOT removes error.

Before:
`#pragma once

namespace dictionary
{

extern template class HashTableInteger<int32_t>;
extern template class HashTableInteger<uint32_t>;
extern template class HashTableInteger<int64_t>;
extern template class HashTableInteger<uint64_t>;
// extern template class HashTableInteger<int128_t>;   // Commented out due to cppcheck limitations.
// extern template class HashTableInteger<uint128_t>;  // Commented out due to cppcheck limitations.

}  // namespace dictionary

namespace std
{
std::string to_string(const uint128_t &);
std::string to_string(const int128_t &);
}  // namespace std
`

After:
`#pragma once

namespace dictionary
{

extern template class HashTableInteger<int32_t>;
extern template class HashTableInteger<uint32_t>;
extern template class HashTableInteger<int64_t>;
extern template class HashTableInteger<uint64_t>;
extern template class HashTableInteger<int128_t>;
extern template class HashTableInteger<uint128_t>;

}  // namespace dictionary

namespace std
{
std::string to_string(const uint128_t &);
std::string to_string(const int128_t &);
}  // namespace std`

Difftastic version 0.37.0 on Ubuntu 22.04 in WSL2.

**Comment by Wilfred** [comment]
Before file:

```
#pragma once

namespace dictionary
{

extern template class HashTableInteger<int32_t>;
extern template class HashTableInteger<uint32_t>;
extern template class HashTableInteger<int64_t>;
extern template class HashTableInteger<uint64_t>;
// extern template class HashTableInteger<int128_t>; // Commented out due to cppcheck limitations.
// extern template class HashTableInteger<uint128_t>; // Commented out due to cppcheck limitations.

} // namespace dictionary

namespace std
{
std::string to_string(const uint128_t &);
std::string to_string(const int128_t &);
} // namespace std
```

The latest released version of difftastic does this:

![Screenshot from 2022-10-28 13-09-13](https://user-images.githubusercontent.com/70800/198723494-b21bf3be-898e-42c2-a014-ce3f8a950730.png)

Whereas trunk does this:

![Screenshot from 2022-10-28 13-09-45](https://user-images.githubusercontent.com/70800/198723526-118d4561-13e6-4a7d-ab0a-0e44824d8f3e.png)

The problem here is that the C++ parser doesn't like this syntax.

```
$ difft --dump-ts one.cpp | grep ERROR
      ERROR (5, 0) - (8, 48)
        ERROR (5, 7) - (5, 48)
            ERROR (5, 16) - (5, 21)
        ERROR (6, 7) - (6, 49)
            ERROR (6, 16) - (6, 21)
        ERROR (7, 7) - (7, 48)
            ERROR (7, 16) - (7, 21)
          ERROR (8, 16) - (8, 21)
```

I suspect this is an upstream parser issue.

**Comment by Wilfred** [comment]
Should be fixed by #472.
